taco-core-server/src/main/java/com/tandanji/taco_core_server/application/ProductsService.java

```diff
@@ -4,24 +4,82 @@
 import com.tandanji.taco_core_server.infrastructure.ProductRepository;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
 
 @Service
 public class ProductsService {
 
-    private ProductRepository productRepository;
+    private final ProductRepository productRepository;
+    private final ValidationService validationService;
 
     @Autowired
-    ProductsService(ProductRepository productRepository) {
+    ProductsService(ProductRepository productRepository, ValidationService validationService) {
         this.productRepository = productRepository;
+        this.validationService = validationService;
+    }
+
+    public void createProduct(Product product, MultipartFile image) throws IOException {
+        validationService.checkValid(product);
+
+        product = saveImage(product, image);
+
+        productRepository.createProduct(product);
     }
 
-    public Product createProduct(Product product) {
-        int result = productRepository.createProduct(product);
+    public static Product saveImage(Product product, MultipartFile image) throws IOException {
+        if (image != null && !image.isEmpty()) {
+            String imageSaveDir = System.getProperty("user.dir") + "/image";
```

Comment:
I'm not sure, but I think the code below should work fine as well instead of assembling directory path manually
```suggestion
            String imageSaveDir = Paths.get(System.getProperty("user.dir"), "image").toString();
```

Reply:
It works well👍 I applied it